homeassistant/components/ozw/climate.py

```diff
@@ -0,0 +1,388 @@
+        if not self.values.temperature:
+            return None
+        return self.values.temperature.value
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported."""
+        if not self.values.operating_state:
+            return None
+        cur_state = self.values.operating_state.value.lower()
+        return HVAC_CURRENT_MAPPINGS.get(cur_state)
+
+    @property
+    def preset_mode(self):
+        """Return preset operation ie. eco, away."""
+        if not self.values.mode:
+            return None
+        # Z-Wave uses mode-values > 10 for presets
+        if self.values.mode.value[VALUE_SELECTED_ID] > 10:
+            return self.values.mode.value[VALUE_SELECTED_LABEL]
+        return PRESET_NONE
+
+    @property
+    def preset_modes(self):
+        """Return the list of available preset operation modes."""
+        if not self.values.mode:
+            return []
+        # Z-Wave uses mode-values > 10 for presets
+        return [PRESET_NONE] + [
+            val[VALUE_LABEL]
+            for val in self.values.mode.value[VALUE_LIST]
+            if val[VALUE_ID] > 10
+        ]
+
+    @property
+    def swing_mode(self):
+        """Return the swing setting."""
+        if not self.values.swing_mode:
+            return None
+        return self.values.swing_mode.value
+
+    @property
+    def swing_modes(self):
+        """Return the list of available swing modes."""
+        if not self.values.swing_mode:
+            return []
+        return [
+            entry[VALUE_LABEL] for entry in self.values.swing_mode.value[VALUE_LIST]
+        ]
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) != 1:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[0].value
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        if len(self._current_mode_setpoint_values) < 2:
+            return None
+        return self._current_mode_setpoint_values[1].value
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature.
+
+        Must know if single or double setpoint.
+        """
+        if len(self._current_mode_setpoint_values) == 1:
+            setpoint = self._current_mode_setpoint_values[0]
+            target_temp = kwargs.get(ATTR_TEMPERATURE)
+            if setpoint is not None and target_temp is not None:
+                setpoint.send_value(target_temp)
+        elif len(self._current_mode_setpoint_values) == 2:
```

Comment:
Where not testing this case, right? Can we add a test of setting temperature with high low set points.

Reply:
done!